test/functional/api/liveupdate_spec.lua

```diff
@@ -0,0 +1,749 @@
+    local f = io.open(filename, 'a')
+    f:write("another line\n")
+    f:flush()
+    f:close()
+
+    -- reopen the file and watch live updates shut down
+    command('edit')
+    expectn('LiveUpdateEnd', {b})
+  end)
+
+  it('allows a channel to watch multiple buffers at once', function()
+    -- edit 3 buffers, make sure they all have windows visible so that when we
+    -- move between buffers, none of them are unloaded
+    local b1, tick1 = editoriginal(true, {'A1', 'A2'})
+    local b1nr = eval('bufnr("")')
+    command('split')
+    local b2, tick2 = open(true, {'B1', 'B2'})
+    local b2nr = eval('bufnr("")')
+    command('split')
+    local b3, tick3 = open(true, {'C1', 'C2'})
+    local b3nr = eval('bufnr("")')
+
+    -- make a new window for moving between buffers
+    command('split')
+
+    command('b'..b1nr)
+    command('normal! x')
+    tick1 = tick1 + 1
+    expectn('LiveUpdate', {b1, tick1, 0, 1, {'1'}})
+    command('undo')
+    tick1 = tick1 + 1
+    expectn('LiveUpdate', {b1, tick1, 0, 1, {'A1'}})
+    tick1 = tick1 + 1
+    expectn('LiveUpdateTick', {b1, tick1})
+
+    command('b'..b2nr)
+    command('normal! x')
+    tick2 = tick2 + 1
+    expectn('LiveUpdate', {b2, tick2, 0, 1, {'1'}})
+    command('undo')
+    tick2 = tick2 + 1
+    expectn('LiveUpdate', {b2, tick2, 0, 1, {'B1'}})
+    tick2 = tick2 + 1
+    expectn('LiveUpdateTick', {b2, tick2})
+
+    command('b'..b3nr)
+    command('normal! x')
+    tick3 = tick3 + 1
+    expectn('LiveUpdate', {b3, tick3, 0, 1, {'1'}})
+    command('undo')
+    tick3 = tick3 + 1
+    expectn('LiveUpdate', {b3, tick3, 0, 1, {'C1'}})
+    tick3 = tick3 + 1
+    expectn('LiveUpdateTick', {b3, tick3})
+  end)
+
+  it('doesn\'t get confused when you turn watching on/off many times',
+     function()
+    local channel = nvim('get_api_info')[1]
+    local b, tick = editoriginal(false)
+
+    -- turn on live updates many times
+    ok(buffer('live_updates_start', b, true))
+    ok(buffer('live_updates_start', b, true))
+    ok(buffer('live_updates_start', b, true))
+    ok(buffer('live_updates_start', b, true))
+    ok(buffer('live_updates_start', b, true))
+    expectn('LiveUpdateStart', {b, tick, origlines, false})
+    eval('rpcnotify('..channel..', "Hello There")')
+    expectn('Hello There', {})
+
+    -- turn live updates off many times
+    ok(buffer('live_updates_stop', b))
+    ok(buffer('live_updates_stop', b))
+    ok(buffer('live_updates_stop', b))
+    ok(buffer('live_updates_stop', b))
+    ok(buffer('live_updates_stop', b))
+    expectn('LiveUpdateEnd', {b})
+    eval('rpcnotify('..channel..', "Hello Again")')
+    expectn('Hello Again', {})
+  end)
+
+  it('is able to notify several channels at once', function()
+    helpers.clear()
+
+    local addsession = function(where)
+      where = alter_slashes(where)
+      print("serverstart('"..where.."')")
+      --eval("serverstart('"..where.."')")
+      helpers.command("call serverstart('"..where.."')")
```

Comment:
Do we really need multiple servers for this? Can't we just connect multiple times to `v:servername` ?

Reply:
Yep, adjusted that.